model/task/task.go

```diff
@@ -994,16 +999,33 @@ func (t *Task) SetAborted(reason AbortInfo) error {
 	)
 }
 
-func (t *Task) SetHasCedarResults(hasCedarResults bool) error {
+// SetHasCedarResults sets the HasCedarResults field of the task to
+// hasCedarResults and, if failedResults is true, sets CedarResultsFailed to
+// true. An error is returned if hasCedarResults is false and failedResults is
+// true as this is an invalid state. Note that if failedResults is false,
+// nothing is set. This is because in cases where separate calls to attach test
+// results are made, only one call needs to have a test failure for the
+// CedarResultsFailed to be set to true.
+func (t *Task) SetHasCedarResults(hasCedarResults, failedResults bool) error {
```

Comment:
If hasCedarResults is false, should we just exit early? Or if we only ever call this with "hasCedarResults=true" we could just omit that parameter

Reply:
We never actually call this with `hasCedarResults==false`. But I made it more flexible on the DB level for testing and whatnot.